Installation failed

`opam pin --switch=~/.local/share/creusot creusot-lsp creusot-ide/ -y` fails with

```
[NOTE] Package creusot-lsp is already pinned to git+file:///home/ebruneton/Programmes/creusot-ide#main (version dev).
[creusot-lsp.dev] synchronised (no changes)
[WARNING] Failed checks on creusot-lsp package definition from source at git+file:///home/ebruneton/Programmes/creusot-ide#main:
  warning 36: Missing field 'bug-reports'
  warning 68: Missing field 'license'
creusot-lsp is now pinned to git+file:///home/ebruneton/Programmes/creusot-ide#main (version dev)

The following actions will be performed:
=== install 1 package
  ∗ creusot-lsp dev (pinned)

<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
[ERROR] The compilation of creusot-lsp.dev failed at "dune build -p creusot-lsp -j 3 @install".

#=== ERROR while compiling creusot-lsp.dev ====================================#
# context     2.3.0 | linux/x86_64 | ocaml.5.3.0 | pinned(git+file:///home/ebruneton/Programmes/creusot-ide#main#1072a05e90a11ec7c75ab9158f6035f424cc9291)
# path        ~/.local/share/creusot/_opam/.opam-switch/build/creusot-lsp.dev
# command     ~/.opam/opam-init/hooks/sandbox.sh build dune build -p creusot-lsp -j 3 @install
# exit-code   1
# env-file    ~/.opam/log/creusot-lsp-103758-dd749e.env
# output-file ~/.opam/log/creusot-lsp-103758-dd749e.out
### output ###
# [...]
# (cd _build/.sandbox/3a5cb6e5bb3e29f6c346afa517219d2f/default && /home/ebruneton/.local/share/creusot/_opam/bin/menhir --explain server/lib/rust_parser.mly --base server/lib/rust_parser --infer-read-reply server/lib/rust_parser__mock.mli.inferred)
# Warning: 2 states have shift/reduce conflicts.
# Warning: 2 shift/reduce conflicts were arbitrarily resolved.
# (cd _build/default && /home/ebruneton/.local/share/creusot/_opam/bin/ocamlopt.opt -w -40 -g -I server/bin/.creusot_lsp.eobjs/byte -I server/bin/.creusot_lsp.eobjs/native -I /home/ebruneton/.local/share/creusot/_opam/lib/ISO8601 -I /home/ebruneton/.local/share/creusot/_opam/lib/angstrom -I /home/ebruneton/.local/share/creusot/_opam/lib/base -I /home/ebruneton/.local/share/creusot/_opam/lib/bas[...]
# File "server/bin/creusot_lsp.ml", line 21, characters 29-62:
# 21 |       let* () = match params.InitializeParams.workspaceFolders with
#                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
# Error: The field "InitializeParams.workspaceFolders" belongs to the record type
#          "Lsp.Types.InitializeParams.t" = "Lsp.Types.InitializeParams.t"
#        but a field was expected belonging to the record type
#          "Linol__.Common_.Lsp.Types.InitializeParams.t" =
#            "Linol__.Common_.Lsp.Types.InitializeParams.t"



<><> Error report <><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
┌─ The following actions failed
│ λ build creusot-lsp dev
└─ 
╶─ No changes have been performed
[NOTE] Pinning command successful, but your installed packages may be out of sync.
```

Creusot itself was previously installed successfully:

`cargo creusot setup status`

```
Creusot installation found.
=== INSTALLATION
Why3:
- path: /home/ebruneton/.local/share/creusot/_opam/bin/why3
- check_version: true
Alt-Ergo:
- mode: builtin
- check_version: true
Z3:
- mode: builtin
- check_version: true
CVC4:
- mode: builtin
- check_version: true
CVC5:
- mode: builtin
- check_version: true
=== PATHS
config: /home/ebruneton/.config/creusot
data:   /home/ebruneton/.local/share/creusot
cache:  /home/ebruneton/.cache/creusot
```

I have opam 2.3.0 with ocaml-compiler 5.3.0, on Ubuntu 24.04.

--- a/.vscode/launch.json
+++ b/.vscode/launch.json
@@ -7,8 +7,8 @@
 			"request": "launch",
 			"name": "Launch Client",
 			"runtimeExecutable": "${execPath}",
-			"args": ["--disable-extensions", "--extensionDevelopmentPath=${workspaceRoot}/vscode"],
-			"outFiles": ["${workspaceRoot}/vscode/out/**/*.js"],
+			"args": ["--disable-extensions", "--extensionDevelopmentPath=${workspaceRoot}"],
+			"outFiles": ["${workspaceRoot}/out/**/*.js"],
 			"preLaunchTask": "Build Server and Extension",
             "env": {
                 "PATH": "${workspaceRoot}/_build/install/default/bin:${env:PATH}"

--- a/README.md
+++ b/README.md
@@ -79,6 +79,11 @@ Available in the command palette (`Ctrl+P`):
 ## Settings
 
 - `creusot.lspPath`: Path to the `creusot-lsp` executable. Default: `""`, finding the executable in Creusot's Opam switch.
+- `creusot.home`, `creusot.dataHome`, `creusot.configHome`: Override the environment variables `HOME`, `XDG_DATA_HOME`, and `XDG_CONFIG_HOME` when invoking Creusot. This allows running Creusot from VS Code and other applications installed via Ubuntu Snap, which sets those variables to some custom directories, breaking Creusot. On Linux you probably want to set these paths:
+
+    + `creusot.home`: `/home/$USER` (where `$USER` is your user name)
+    + `creusot.dataHome`: `/home/$USER/.local/share` (there should be a `creusot` directory at this location)
+    + `creusot.configHome`: `/home/$USER/.config` (there should be a `creusot` directory at this location)
 
 ## Developers' corner
 

--- a/package.json
+++ b/package.json
@@ -26,6 +26,21 @@
           "type": "string",
           "default": "",
           "description": "Path to the creusot-lsp executable"
+        },
+        "creusot.home": {
+          "type": "string",
+          "default": "",
+          "description": "Override HOME environment variable"
+        },
+        "creusot.dataHome": {
+          "type": "string",
+          "default": "",
+          "description": "Override XDG_DATA_HOME environment variable"
+        },
+        "creusot.configHome": {
+          "type": "string",
+          "default": "",
+          "description": "Override XDG_CONFIG_HOME environment variable"
         }
       }
     },

--- a/package.json
+++ b/package.json
@@ -90,14 +90,6 @@
         "type": "creusot"
       }
     ],
-    "menus": {
-      "editor/context": [
-        {
-          "command": "creusot.why3ide",
-          "group": "2_creusot"
-        }
-      ]
-    },
     "grammars": [
       {
         "language": "coma",

--- a/vscode/src/extension.ts
+++ b/vscode/src/extension.ts
@@ -20,13 +20,29 @@ function getDefaultLSPPath(): string {
   }
 }
 
+function creusotLspPath(): string {
+   return workspace.getConfiguration("creusot").get("lspPath") || ""
+}
+
+function creusotHome(): string {
+   return workspace.getConfiguration("creusot").get("home") || ""
+}
+
+function creusotDataHome(): string {
+   return workspace.getConfiguration("creusot").get("dataHome") || ""
+}
+
+function creusotConfigHome(): string {
+   return workspace.getConfiguration("creusot").get("configHome") || ""
+}
+
+function getLspPath(): string {
+  const lspPath: string = creusotLspPath();
+  return lspPath === "" ? getDefaultLSPPath() : lspPath;
+}
+
 function getServerExecutable(context): Executable {
-  const lspPath: string | undefined = workspace.getConfiguration("creusot").get("lspPath");
-  if (lspPath === undefined || lspPath === "") {
-    return { command: getDefaultLSPPath() };
-  } else {
-    return { command: lspPath };
-  }
+  return { command: getLspPath() }
 }
 
 const languages = [
@@ -116,15 +132,32 @@ async function createTests(client: LanguageClient) {
           // Save the rust file
           if (test.uri) { await vscode.workspace.save(test.uri); }
           process.chdir(rootPath);
+          const options: child_process.SpawnSyncOptions = {};
+          const home = creusotHome();
+          const dataHome = creusotDataHome();
+          const configHome = creusotConfigHome();
+          if (home !== "" || dataHome !== "" || configHome !== "") {
+            const env = { ...process.env };
+            if (home !== "") {
+              env.HOME = home;
+            }
+            if (dataHome !== "") {
+              env.XDG_DATA_HOME = home;
+            }
+            if (configHome !== "") {
+              env.XDG_CONFIG_HOME = home;
+            }
+            options.env = env;
+          }
           // cargo creusot
-          const buildOutput = child_process.spawnSync("cargo", ["creusot"]);
+          const buildOutput = child_process.spawnSync("cargo", ["creusot"], options);
           if (buildOutput.status !== 0) {
             const logs = buildOutput.stdout.toString() + "\n" + buildOutput.stderr.toString();
-            run.failed(test, new vscode.TestMessage("'cargo creusot' failed\n" + logs));
+            run.failed(test, new vscode.TestMessage("Failed translation\n" + logs));
             return;
           }
           // why3find prove
-          const output = child_process.spawnSync("cargo", ["creusot", "prove", test.id]);
+          const output = child_process.spawnSync("cargo", ["creusot", "prove", test.id], options);
           run.appendOutput(`Finishing test ${test.label}\n\r`);
           if (output.status !== 0) {
             const logs = output.stdout.toString() + "\n" + output.stderr.toString();

